presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestSpatialJoinOperator.java

```diff
@@ -210,6 +210,7 @@ public void testYield()
 
         RowPagesBuilder buildPages = rowPagesBuilder(ImmutableList.of(GEOMETRY, VARCHAR))
                 .row(POLYGON_A, "A")
+                .pageBreak()
```

Comment:
I understand you need 2 pages for any useful test with addresses. Did you manage to craft a test that would exploit the bug within reasonable time?  Maybe 3,000,000,000 identical (shared) geometries would do that?

Reply:
This bug happens whenever join has addition function and a record from a build page other then first matches the spatial condition. toIntExact(joinAddress) fails with "integer overflow" whenever joinAddress encodes a record on a page other than first. Hence, this simple change was sufficient to reproduce the issue.